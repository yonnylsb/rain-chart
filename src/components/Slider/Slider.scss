@import '../../common/styles/colors';

.slider {
    &-label {
        display: block;
        font-size: 35px;
        text-align: center;
        margin-bottom: 60px;
        color: $blue-1;
    }
    &-input {
        overflow: hidden;
        -webkit-appearance: none;
        background-color: $gray-2;
        height: 20px;
        width: 100%;
        border-radius: 20px;

        &:hover {
            opacity: 1;
        }

        &:focus {
            outline: 0 none;
        }
          
        &::-webkit-slider-thumb {
            width: 20px;
            height: 20px;
            cursor: ew-resize;
            background: $gray-3;
            border-radius: 20px;
            top: 0;
            box-shadow: -3000px 0px 0px 2990px $blue-2;
            -webkit-appearance: none;
        }
          
        &::-moz-range-thumb {
            width: 20px;
            height: 20px;
            border-radius: 20px;
            background: #c4c4c4;
            cursor: pointer;
            border: 0 none;
            box-shadow: -3000px 0px 0px 2990px $blue-2;
        }
    }
    &-value {
        display: block;
        text-align: center;
        font-size: 100px;
        color: $blue-1
    }
}