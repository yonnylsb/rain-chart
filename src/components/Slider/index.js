import React, {useState, useEffect} from 'react';
import './Slider.scss'

function Slider(props){
    const calcDefault = props.max - Math.floor((props.max - props.min)/2);
    const [sliderValue, setSliderValue] = useState(calcDefault || 0);

    useEffect(()=>{
        props.onSliderChange(calcDefault);
    }, []);

    return (
        <form className="slider">
            <label
                className="slider-label"
                htmlFor={props.identifier}
            >
                {props.title}
            </label>
            <input 
                id={props.identifier}
                className="slider-input"
                type="range"
                min={props.min}
                max={props.max}
                step="1"
                onChange={onChange}
                value={sliderValue}
            />
            <div
                className="slider-value"
            >
                {sliderValue}
            </div>
        </form>
    )

    function onChange(e){
        let value = e.target.value;

        setSliderValue(value);

        if(props.onSliderChange){
            props.onSliderChange(value);
        }
    }
}


export default Slider;