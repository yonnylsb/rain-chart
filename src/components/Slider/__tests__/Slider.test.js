import React from 'react';
import renderer from 'react-test-renderer';
import Slider from '../index';

test('Slider should render', () => {
    const component = renderer.create(
        <Slider
            onSliderChange={()=>{}}
        ></Slider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Slider should render with props', () => {
    const component = renderer.create(
        <Slider
            title="title Test"
            onSliderChange={()=>{}}
            min={0}
            max={200}
        ></Slider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
