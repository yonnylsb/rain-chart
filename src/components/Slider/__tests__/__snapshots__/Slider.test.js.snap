// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render 1`] = `
<form
  className="slider"
>
  <label
    className="slider-label"
  />
  <input
    className="slider-input"
    onChange={[Function]}
    step="1"
    type="range"
    value={0}
  />
  <div
    className="slider-value"
  >
    0
  </div>
</form>
`;

exports[`Slider should render with props 1`] = `
<form
  className="slider"
>
  <label
    className="slider-label"
  >
    title Test
  </label>
  <input
    className="slider-input"
    max={200}
    min={0}
    onChange={[Function]}
    step="1"
    type="range"
    value={100}
  />
  <div
    className="slider-value"
  >
    100
  </div>
</form>
`;
